cura/Settings/CuraContainerStack.py

```diff
@@ -387,6 +389,18 @@ def getExtruderPositionValueWithDefault(self, key):
             value = int(Application.getInstance().getMachineManager().defaultExtruderPosition)
         return value
 
+    def getProperty(self, key: str, property_name: str, context = None) -> Any:
+        if property_name == "settable_per_extruder":
+            # Setable per extruder isn't a value that can ever change. So once we requested it once, we can just keep
+            # that in memory.
+            try:
```

Comment:
You could alternatively use:
```
if key in self._settable_per_extruder_cache:
    return self._settable_per_extruder_cache[key]
else:
    ......
```
This will give you a 20% speed increase but for cases that the key isn't present. for cases that the key is present your code is  25% faster. So the question remains do we query more for existing keys then non-existing keys?

Reply:
Check the description of the commit :)